Platform/RaptorlakeBoardPkg/BoardConfigRplp.py

```diff
@@ -0,0 +1,703 @@
+            self.UEFI_VARIABLE_SIZE = 0x00040000
+        self.UCODE_SIZE           = 0x000B7000
+        self.MRCDATA_SIZE         = 0x00010000
+        self.CFGDATA_SIZE         = 0x00004000
+        self.KEYHASH_SIZE         = 0x00001000
+        self.VARIABLE_SIZE        = 0x00002000
+        self.SBLRSVD_SIZE         = 0x00001000
+        self.FWUPDATE_SIZE        = 0x00020000 if self.ENABLE_FWU else 0
+
+        self.SIIPFW_SIZE = 0x1000
+
+        # Declaring UCODE_SLOT_SIZE populates the uCode region with as many
+        # uCode slots as possible, even if there aren't enough uCode patches
+        # to fill each slot. This can have a slight performance impact on SBL,
+        # as it tries to load dummy uCode patches.
+        self.UCODE_SLOT_SIZE      = 0x4C000
+
+        # If ENABLE_SBL_RESILIENCY is 1, BiosRedAssistance FIT strap setting
+        # needs to be manually changed to Enabled and TopSwapOverride
+        # flash setting needs to be manually changed to 4MB in stitch config
+        self.ENABLE_SBL_RESILIENCY = 0
+        self.BUILD_IDENTICAL_TS    = 0
+
+        if self.ENABLE_SBL_RESILIENCY:
+            self.BUILD_IDENTICAL_TS    = 1
+
+        if self.BUILD_IDENTICAL_TS:
+            self.TOP_SWAP_SIZE         = 0x400000
+            self.REDUNDANT_SIZE        = self.STAGE2_SIZE + self.FWUPDATE_SIZE + self.CFGDATA_SIZE + \
+                                         self.KEYHASH_SIZE
+        else:
+            self.TOP_SWAP_SIZE        = 0x80000
+            self.REDUNDANT_SIZE       = self.UCODE_SIZE + self.STAGE2_SIZE + self.STAGE1B_SIZE + \
+                                        self.FWUPDATE_SIZE + self.CFGDATA_SIZE + self.KEYHASH_SIZE
+
+        self.FUSA_SUPPORT         = 0
+        self.FUSA_SIZE            = 0
+
+        # These can be overridden by FUSA .dlt file.
+        self.ENABLE_TCC           = 0
+        self.ENABLE_TSN           = 0
+
+        if self.FUSA_SUPPORT:
+            self.ENABLE_PRE_OS_CHECKER = 0
+            if self.ENABLE_PRE_OS_CHECKER:
+                self.POSC_SIZE = 0x00080000
+                self.SIIPFW_SIZE += self.POSC_SIZE
+            self.FUSA_BIST_PATTERN_SUPPORT  = 0
+            if self.FUSA_BIST_PATTERN_SUPPORT:
+                self.FUSA_PATTERN_IS_FFS        = 0
+                self.FUSA_STARTUP_BIST_FILE     = ''
+                self._FUSA_PERIODIC_BIST_FILE    = ''
+                self.FSBP_SIZE        = 0x1000000
+                self.FPBP_SIZE        = 0x1000000
+                self.FUSA_SIZE        = 0x1000 + self.FPBP_SIZE + self.FSBP_SIZE
+
+            # Allow TCC to be enabled by FuSa dlt file
+            FusaConfig = ''
+            brd_cfg_src_dir = os.path.join(os.environ['SBL_SOURCE'], 'Platform', self.BOARD_PKG_NAME, 'CfgData')
+            brd_cfg2_src_dir = '.'
+            if hasattr(self, 'BOARD_PKG_NAME_OVERRIDE'):
+                brd_cfg2_src_dir = os.path.join(os.environ['PLT_SOURCE'], 'Platform', self.BOARD_PKG_NAME_OVERRIDE, 'CfgData')
```

Comment:
PLT_SOURCE should be changed to SBL_SOURCE, otherwise, build failure happened when FUSA_SUPPORT is set to 1

Reply:
I think this should be ok. brd_cfg2_src_dir should only be used in the event the file doesn't exist under brd_cfg_src_dir. I am not seeing a build error in this case. Do you?

Reply:
It's probably because I built the code directly with BuildLoader.py instead of SblBuild.py

Reply:
Error log
 File "BuildLoader.py", line 1458, in main
    if brdcfg.Board().BOARD_NAME:
  File "C:\projects\sbl\slimbootloader\Platform\RaptorlakeBoardPkg\BoardConfigRplp.py", line 223, in __init__
    brd_cfg2_src_dir = os.path.join(os.environ['PLT_SOURCE'], 'Platform', self.BOARD_PKG_NAME_OVERRIDE, 'CfgData')
  File "C:\Users\vincentc\AppData\Local\Programs\Python\Python38\lib\os.py", line 675, in __getitem__
    raise KeyError(key) from None
KeyError: 'PLT_SOURCE'

Reply:
Ah, I understand. This is fixed now.